src/Sulu/Bundle/WebsiteBundle/Resources/public/js/services/reference-store.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define(function() {
+
+    'use strict';
+
+    var synonyms = {},
+        references = {},
+
+        resolveAlias = function(alias) {
```

Comment:
the provider alias will be used to register the ids. this will not work if we have multiple aliases (in the backend the provider itself register it). In the javascript we have no providers here the alias will be used to register the id and then we have to know all the aliases here to be sure that they dont get lost.